packages/flutter_tools/lib/src/fuchsia/fuchsia_device.dart

```diff
@@ -657,11 +727,26 @@ class FuchsiaDevice extends Device {
     return ports;
   }
 
+  /// Determine if the fuchsia device is running a workstation build.
+  ///
+  /// If the device is running a workstation build, `session_control` should be
+  /// used to launch apps, otherwise `tiles_ctl` should be used.
+  Future<bool> isWorkstation() async {
+    final RunResult result = await shell('cat /config/build-info/product');
+    if (result.exitCode != 0) {
+      return false;
+    }
+    if (result.stdout.contains('workstation')) {
```

Comment:
This is not part of the platform API and should not be relied upon. This will break if workstation changes their product name and it will also not scale if we add other products.